### PR TITLE
Improved console output

### PR DESCRIPTION
**Pull request by evilny0**
I added hashrate per card, and colors for accepted/refused shares. Only tested it on CUDA & Nicehash, so I hope I did not miss anything.

### CODE REVIEW

File: libethcore/Miner.h [hunk 1]
```diff
@@ -81,6 +81,31 @@ inline std::ostream& operator<<(std::ostream& _out, WorkingProgress _p)
 	return _out;
 }
 
+struct FarmWorkingProgress
```
Comment (chfast) [inline libethcore/Miner.h]: I don't like splitting this single struct into two. Can we keep it as one?

**Comment by derubm** [comment]
if you guys work on the console output, can you eventually add 
1. a mh/s on "Submitted and accepted", and 
2. showing the Nonce in Hexadecimal, not in decimal ? 

for .1 (in -v 1 mode ethminer only shows information and errors) but on submit at least i ´d like to know onwhat hashing speed the card was working)
for .2 Nonce is shown in the blockbrowser as decimal, in ethminer as decimal)

File: libethcore/Miner.h (hunk 1 above)
Comment (evilny0) [inline libethcore/Miner.h]: @chfast  You mean something like this : https://gist.github.com/evilny0/19f2172a7f45a3b00bfd00ec0e8290c2 ?

**Comment by LeRoyJulian** [comment]
When will this be merged ?

**Comment by EoD** [comment]
@LeRoyJulian you can try it at https://ci.appveyor.com/project/ethereum-mining/ethminer/build/259/job/bmphpg69qhsoeut6/artifacts

**Comment by derubm** [comment]
i can not see any colored output on that .. ? (Win 10 / DOS)

**Comment by evilny0** [comment]
I added colors, but colors are not supported on Windows version, so it only works on Linux.

I could try to look at how to implement colors on Windows terminal (or powershell)... but I could only test on a VM, so no GPU to run the miner on !

**Comment by derubm** [comment]
that would work on windows:
` HANDLE hConsole = GetStdHandle(STD_OUTPUT_HANDLE);
 for(int k = 1; k < 255; k++)
  {
    SetConsoleTextAttribute(hConsole, k);
    cout << k << " Colors! " << endl;
  }`

**Comment by evilny0** [comment]
I added colors for ANSI terminals on Windows.

I tried to add it for non-ANSI terminals, and succeeded to some extent... when we write logs directly to the console. However, the vast majority of logs are first built, and written to the console later on. Which mean changing the color when the logs are built has no effect on the console output. Changing this means probably means a major rework of the logs, and I'm not going to do that :).

**Comment by Blacksuu** [comment]
@evilny0 well that last try what you done with "Fixed missing static keyword." it generate bunch of "m" from time to time aka it mess thing :) noo colour change only a extra field line like total ... gpu0 value ... i test if under Windows 7 Service Pack 1 (6.1.7601) 64bit

**Comment by evilny0** [comment]
Yes, I noticed. I did a few tries on Linux with no problem before commit, but now I have issues. I'll fix it.

**Comment by evilny0** [comment]
About colors, as I said it's only if the terminal supports ANSI escape codes. On Windows, it basically means Windows 10 only. And it doesn't seem to be all versions, since on the VM I have, it's not supported... it works in Powershell (Win10) though.

**Comment by Blacksuu** [comment]
yup last build you done noo more bunch of "m", noo colour on win7 but i undestend reading previos comment no biggie, my sugestion if is possible to be add a total share counter like:
> ETH - Total Speed: X Mh/s, Total Shares: X, Rejected: X, Stale: X <
cheers

**Comment by evilny0** [comment]
I'm not sure I understand. It's already there. Shares are here : [A2+0:R0+0:F0], and total is on the next line. Example :

  m  00:36:25|ethminer  Mining on #b04b76ab… [A2+0:R0+0:F0]
  m  00:36:25|ethminer  Total: 145.43MH/s GPU0: 24.96MH/s GPU1: 24.71MH/s GPU2: 24.20MH/s GPU3: 24.20MH/s GPU4: 23.69MH/s GPU5: 23.69MH/s

**Comment by Blacksuu** [comment]
my sugestion was to add a counter what to end like this:
m 00:36:25|ethminer Total: 145.43MH/s, Total Shares: 200, Rejected: 5, Stale: 2, GPU0: 24.96MH/s GPU1: 24.71MH/s GPU2: 24.20MH/s GPU3: 24.20MH/s GPU4: 23.69MH/s GPU5: 23.69MH/s

**Comment by evilny0** [comment]
I'll split this PR to several. Colors for windows & hashrate per GPU should be in a different one.

**Comment by evilny0** [comment]
I created #225 & #226 instead of this one.

**Pull request closed without merging**